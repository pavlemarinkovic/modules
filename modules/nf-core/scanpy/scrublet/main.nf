process SCRUBLET {
    tag "$meta.id"
    label 'process_medium'
    label 'process_gpu'

    conda "${moduleDir}/environment.yml"
    container "${ task.ext.use_gpu ? 'ghcr.io/scverse/rapids_singlecell:v0.10.8' :
        workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/scanpy_scikit-image:185956e1b73ad93d':
        'community.wave.seqera.io/library/scanpy_scikit-image:066cc4cb329a805c' }"

    input:
    tuple val(meta), path(h5ad)

    output:
    tuple val(meta), path("*.h5ad"), emit: h5ad
    tuple val(meta), path("*.pkl") , emit: predictions
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    template 'scrublet.py'

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def use_gpu = task.ext.use_gpu ?: "false"

    """
    touch ${prefix}.h5ad
    touch ${prefix}.pkl

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python3 -c "import platform; print(platform.python_version())")
        scanpy: \$(python3 -c "import scanpy as sc; print(sc.__version__)")
    END_VERSIONS
    """
}
