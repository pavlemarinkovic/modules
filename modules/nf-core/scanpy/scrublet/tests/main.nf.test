nextflow_process {

    name "Test Process SCRUBLET"
    script "../main.nf"
    process "SCRUBLET"

    tag "modules"
    tag "modules_nfcore"
    tag "scanpy"
    tag "scanpy/scrublet"


    test("scrublet_normal") {

        when {
            process {
                """
                input_file_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/scdownstream/samples/SAMN14430799_custom_emptydrops_filter_matrix.h5ad'

                input[0] = Channel.of([
                    [ id:'test' ],
                    file(input_file_path, checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.h5ad).match("h5ad") },
                { assert snapshot(process.out.predictions).match("predictions") },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }
    }
}
