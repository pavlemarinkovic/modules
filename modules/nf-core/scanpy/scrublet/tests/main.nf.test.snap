{
    "h5ad": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.h5ad:md5,fa8725968d10852abd22d7fbb3b757d0"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T11:47:45.524343"
    },
    "h5ad_stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T11:48:00.636666"
    },
    "versions_stub": {
        "content": [
            [
                "versions.yml:md5,795de5a653bf3624043fc660b261912f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T11:48:00.860557"
    },
    "predictions_stub": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.pkl:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T11:48:00.747565"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,11b0b1f5180aeb7d5de3ed97694cc88b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T11:47:45.769802"
    },
    "predictions": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.pkl:md5,9e8f722da00ae084ca839c36809e60d8"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T11:47:45.656588"
    }
}